File: web/tests/reference/matrix/set-matrix-zeroes.py
def set_zeroes(matrix):
    rows, cols = len(matrix), len(matrix[0])
    first_col_zero = False

    for r in range(rows):
        if matrix[r][0] == 0:
            first_col_zero = True
        for c in range(1, cols):
            if matrix[r][c] == 0:
                matrix[r][0] = 0
                matrix[0][c] = 0

    for r in range(1, rows):
        for c in range(1, cols):
            if matrix[r][0] == 0 or matrix[0][c] == 0:
                matrix[r][c] = 0

    if matrix[0][0] == 0:
        for c in range(cols):
            matrix[0][c] = 0

    if first_col_zero:
        for r in range(rows):
            matrix[r][0] = 0

File: web/tests/reference/matrix/set-matrix-zeroes.js
function setZeroes(matrix) {
  const rows = matrix.length;
  const cols = matrix[0].length;
  let firstColZero = false;

  for (let r = 0; r < rows; r++) {
    if (matrix[r][0] === 0) firstColZero = true;
    for (let c = 1; c < cols; c++) {
      if (matrix[r][c] === 0) {
        matrix[r][0] = 0;
        matrix[0][c] = 0;
      }
    }
  }

  for (let r = 1; r < rows; r++) {
    for (let c = 1; c < cols; c++) {
      if (matrix[r][0] === 0 || matrix[0][c] === 0) matrix[r][c] = 0;
    }
  }

  if (matrix[0][0] === 0) {
    for (let c = 0; c < cols; c++) matrix[0][c] = 0;
  }

  if (firstColZero) {
    for (let r = 0; r < rows; r++) matrix[r][0] = 0;
  }
}
